h/services/annotation_moderation.py

```diff
@@ -78,6 +102,129 @@ def update_status(self, action: AnnotationAction, annotation: Annotation) -> Non
 
         self.set_status(annotation, new_status)
 
+    def queue_moderation_change_email(self, request, moderation_log_id: int) -> None:
+        """Queue an email to be sent to the user about moderation changes on their annotations."""
+
+        moderation_log = self._session.get(ModerationLog, moderation_log_id)
+
+        annotation = moderation_log.annotation
+        group = annotation.group
+        author = self._user_service.fetch(annotation.userid)
+
+        if not group.pre_moderated:
+            # We'll start only sending these emails for pre-moderated groups
+            # For now this ties these emails to the FF for moderation emails
+            return
+
+        if not author or not author.email:
+            # We can't email the user if we don't have an email for them.
+            return
+
+        # If there is no active 'moderated' subscription for the user being mentioned.
+        if not self._subscription_service.get_subscription(
+            user_id=author.userid, type_=Subscriptions.Type.MODERATED
+        ).active:
+            return
+
+        old_status = moderation_log.old_moderation_status
+        new_status = moderation_log.new_moderation_status
+
+        # These are the transitions that will trigger an email to be sent
+        email_sending_status_changes = {
+            (ModerationStatus.PENDING, ModerationStatus.APPROVED),
+            (ModerationStatus.PENDING, ModerationStatus.DENIED),
+            (ModerationStatus.APPROVED, ModerationStatus.PENDING),
+            (ModerationStatus.APPROVED, ModerationStatus.DENIED),
+            (ModerationStatus.DENIED, ModerationStatus.APPROVED),
+            (ModerationStatus.SPAM, ModerationStatus.APPROVED),
+        }
+        if (old_status, new_status) not in email_sending_status_changes:
+            return
+
+        template_base = "h:templates/emails/moderated_annotation_notification"
+        context = {
+            "user_display_name": author.display_name or f"@{author.username}",
+            "annotation_url": links.incontext_link(request, annotation)
+            or request.route_url("annotation", id=annotation.id),
+            "annotation": annotation,
+            "annotation_quote": annotation.quote,
+            "app_url": request.registry.settings.get("h.app_url"),
+            "unsubscribe_url": request.route_url(
+                "unsubscribe",
+                token=self._subscription_service.get_unsubscribe_token(
+                    user_id=author.userid, type_=Subscriptions.Type.MODERATED
+                ),
+            ),
+            "preferences_url": request.route_url("account_notifications"),
```

Comment:
This URL still doesn't list this type of email, added a card to the board about it.